### PR TITLE
non-blocking subscription receive

### PR DESCRIPTION
**Pull request by dwsutherland**
This is a small change with no associated Issue.
(One reviewer should do)

Sibling PR to https://github.com/cylc/cylc-uiserver/pull/135
(This one needs to go in first)

Useful as non-blocking receive on subscription won't hang on stoppage.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Already covered by existing tests.
- [x] No change log entry required.
- [x] No documentation update required.

### CODE REVIEW

**Comment by hjoliver** [comment]
@dwsutherland - why do we need to keep the blocking version as well, as an option?

**Comment by dwsutherland** [comment]
> @dwsutherland - why do we need to keep the blocking version as well, as an option?

Because it's faster (?)... We don't "need" to ... Fine, i'll drop it.

**Comment by hjoliver** [comment]
> > @dwsutherland - why do we need to keep the blocking version as well, as an option?
> 
> Because it's faster (?)... We don't "need" to ... Fine, i'll drop it.

No that's cool, we can leave it there in case it's useful in future.  I was just wondering if you had a purpose in mind...

**Comment by dwsutherland** [comment]
> > > @dwsutherland - why do we need to keep the blocking version as well, as an option?
> > 
> > 
> > Because it's faster (?)... We don't "need" to ... Fine, i'll drop it.
> 
> No that's cool, we can leave it there in case it's useful in future. I was just wondering if you had a purpose in mind...

Already done! If we need the other version we can always reinstate it..

File: cylc/flow/network/subscriber.py [hunk 1]
```diff
@@ -95,8 +98,10 @@ async def subscribe(self, msg_handler, *args, **kwargs):
             if self.stopping:
                 break
             try:
-                [topic, msg] = await self.socket.recv_multipart()
+                [topic, msg] = await self.socket.recv_multipart(
+                    flags=zmq.NOBLOCK)
             except zmq.error.ZMQError:
+                await asyncio.sleep(NO_RECEIVE_INTERVAL)
```
Comment (oliver-sanders) [inline cylc/flow/network/subscriber.py:104]: Does setting the ZMQ timeout not work with the PUB-SUB model?

Comment (dwsutherland) [reply cylc/flow/network/subscriber.py:104]: Even if it did, this is a legitimate/documented approach:
https://pyzmq.readthedocs.io/en/latest/api/zmq.html#zmq.Socket.recv

So if there's a reason to shift to an alternative we can, but until then, this is good to go IMO.

**Pull request merged**